### PR TITLE
HEALPix zarr dataloader options (persistent_workers, prefetch_factor, in_order, drop_last fix)

### PR DESCRIPTION
**Pull request by yikwill**
# PhysicsNeMo Pull Request

## Description
Expose PyTorch `DataLoader` tuning knobs on HEALPix zarr data modules (`TimeSeriesDataModuleZarr` and `CoupledTimeSeriesDataModuleZarr`):

- `persistent_workers` — keep workers alive between epochs (only when `num_workers > 0`)
- `prefetch_factor` — batches prefetched per worker (optional; only when `num_workers > 0`)
- `in_order` — optional passthrough to `DataLoader` when supported by the installed PyTorch
- **`drop_last` fix** — honor the module's `drop_last` config for train/val loaders instead of hard-coding `True`; keep `drop_last=False` for test

All new kwargs default to backwards-compatible values (`persistent_workers=False`, `prefetch_factor=None`, `in_order=None`).

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

## Dependencies

- No hard dependency on other open PRs (touches only `data_modules_zarr.py`).
- Independent of HEALPix model / loss feature branches.

### CODE REVIEW

File: physicsnemo/datapipes/healpix/data_modules_zarr.py [hunk 1]
```diff
@@ -62,10 +62,13 @@ def __init__(
         add_insolation: bool = False,
         num_workers: int = 4,
         pin_memory: bool = True,
+        persistent_workers: bool = False,
+        prefetch_factor: Optional[int] = None,
         forecast_init_times: Optional[Sequence] = None,
         add_train_noise: Optional[bool] = False,
         train_noise_params: Optional[DictConfig] = None,
         train_noise_seed: Optional[int] = 42,
+        in_order: Optional[bool] = None,
```
Comment (daviddpruitt) [inline physicsnemo/datapipes/healpix/data_modules_zarr.py:71]: Do we have a use case for this? I don't remember this being an issue.

File: physicsnemo/datapipes/healpix/data_modules_zarr.py [hunk 2]
```diff
@@ -353,7 +371,7 @@ def train_dataloader(self, num_shards=1, shard_id=0) -> DataLoader:
             num_shards=num_shards,
             shard_id=shard_id,
             shuffle=self.shuffle,
-            drop_last=True,
+            drop_last=self.drop_last,
```
Comment (daviddpruitt) [inline physicsnemo/datapipes/healpix/data_modules_zarr.py:374]: I recall that we set this to True to avoid weirdness around batches that didn't fit the full rollout. I have no objection to making it user selectable though.

**Review by daviddpruitt — APPROVED** [review]

File: physicsnemo/datapipes/healpix/data_modules_zarr.py (hunk 1 above)
Comment (yikwill) [reply physicsnemo/datapipes/healpix/data_modules_zarr.py:71]: For reasons I don't entirely understand, I found that it decreased training speed jitter on Perlmutter when training across many nodes and using a Zarr 3 dataset (which I use for some of my aquaplanet work). I don't think most people will use it, but I don't see harm in adding the option with the default as None.
